### PR TITLE
Add proto and preferences datastore functionality

### PR DESCRIPTION
**Pull request by SofiaZarzabal**
## Description

- Add view model, proto, preferences and protobuf dependencies
- Update the view of preferences and proto fragments
- Creation of their respective view models
- Creation of `user_data` proto file

## Evidences
The evidences will be attached to the Slack channel due to the size of the videos.

### CODE REVIEW

File: app/src/main/java/com/example/datastoreapp/fragment/PreferencesDataStoreFragment.kt [hunk 1]
```diff
@@ -3,17 +3,54 @@ package com.example.datastoreapp.fragment
...
+                }
+            }
+        }
+        configView()
+    }
+
+    private fun configView() = with(binding) {
+        btnSave.setOnClickListener {
+            viewModel?.saveInfoPressed(
+                if (ageEditText.text.toString().isEmpty()) -1
```
Comment (SartiMau) [inline app/src/main/java/com/example/datastoreapp/fragment/PreferencesDataStoreFragment.kt]: this logic should be in the viewmodel so you will be able to test it.

File: app/src/main/java/com/example/datastoreapp/fragment/PreferencesDataStoreFragment.kt [hunk 2]
```diff
@@ -3,17 +3,54 @@ package com.example.datastoreapp.fragment
...
+                else ageEditText.text.toString().toInt()
+            )
+            ageEditText.text?.clear()
+            if (tvShowInfo.isVisible) tvShowInfo.visibility = GONE
```
Comment (SartiMau) [inline app/src/main/java/com/example/datastoreapp/fragment/PreferencesDataStoreFragment.kt]: please use curly braces

File: app/src/main/java/com/example/datastoreapp/fragment/ProtoDataStoreFragment.kt [hunk 3]
```diff
@@ -3,17 +3,63 @@ package com.example.datastoreapp.fragment
...
+                            user.age.toString()
+                        )
+                    }
+                }
+            }
+        }
+        configView()
+    }
+
+    private fun configView() = with(binding) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/datastoreapp/fragment/ProtoDataStoreFragment.kt:48]: apply the same comments from PreferencesDataStoreFragment.kt, please

File: app/src/main/java/com/example/datastoreapp/viewmodel/ProtoDataStoreViewModel.kt [hunk 4]
```diff
@@ -0,0 +1 @@
...
                userInfoMutableLiveData.postValue(
                    UserInfo(
                        it.userName,
                        it.userAge
                    )
                )
            }
        }
    }
}
```
Comment (SartiMau) [inline app/src/main/java/com/example/datastoreapp/viewmodel/ProtoDataStoreViewModel.kt]: is this file okay?

File: app/src/main/java/com/example/datastoreapp/UserInfo.kt [hunk 5]
```diff
@@ -0,0 +1 @@
+package com.example.datastoreappdata class UserInfo(    val name: String,    val age: Int)
```
Comment (SartiMau) [inline app/src/main/java/com/example/datastoreapp/UserInfo.kt:1]: Is this file okay? I never seen before de ^M

**Review by SartiMau — CHANGES_REQUESTED** [review]
To avoid pushing the .idea files you should update the gitignore file. You can copy the correct gitignore file from here https://drive.google.com/file/d/132AvoWJgl9XLv5Ok7MD0PrYAOQ0MvMQ1/view?usp=sharing

File: app/src/main/java/com/example/datastoreapp/storetype/ProtoDataStoreSerializer.kt [hunk 6]
```diff
@@ -0,0 +1 @@
...
    }

    override suspend fun writeTo(t: UserData, output: OutputStream) =
        withContext(Dispatchers.IO) { t.writeTo(output) }
}

val Context.createUserDataStore: DataStore<UserData> by dataStore(
    fileName = "user_data.pb",
    serializer = ProtoDataStoreSerializer
)
```
Comment (FranciscoGaraventa) [inline app/src/main/java/com/example/datastoreapp/storetype/ProtoDataStoreSerializer.kt:1]: I think you are having issues with the line endings on kt files. You can modified it on Settings > Code Style > Kotlin > Tabs and Indents

File: app/src/main/java/com/example/datastoreapp/storetype/PreferencesDataStore.kt [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+    suspend fun setAge(age: Int) {
+        dataStore.edit { preferences -> preferences[pdsAge] = age }
+    }
+
+    companion object {
+        private const val PDS_NAME = "PDSStore"
+        private const val PDS_AGE_KEY = "PDS_AGE_KEY"
+    }
+}
```
Comment (FranciscoGaraventa) [inline app/src/main/java/com/example/datastoreapp/storetype/PreferencesDataStore.kt]: EOF

File: app/src/main/java/com/example/datastoreapp/viewmodel/PreferencesDataStoreViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    fun loadInfoPressed() {
+        viewModelScope.launch(Dispatchers.IO) {
+            preferencesDataStoreStorage.getAge().collect {
+                mutableLiveData.postValue(it)
+            }
+        }
+    }
+}
```
Comment (FranciscoGaraventa) [inline app/src/main/java/com/example/datastoreapp/viewmodel/PreferencesDataStoreViewModel.kt]: EOF

File: app/src/main/java/com/example/datastoreapp/fragment/PreferencesDataStoreFragment.kt (hunk 1 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/datastoreapp/fragment/PreferencesDataStoreFragment.kt]: Fixed!

File: app/src/main/java/com/example/datastoreapp/fragment/PreferencesDataStoreFragment.kt (hunk 2 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/datastoreapp/fragment/PreferencesDataStoreFragment.kt]: Fixed!

File: app/src/main/java/com/example/datastoreapp/fragment/ProtoDataStoreFragment.kt (hunk 3 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/datastoreapp/fragment/ProtoDataStoreFragment.kt:48]: Fixed!

File: app/src/main/java/com/example/datastoreapp/storetype/PreferencesDataStore.kt (hunk 7 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/datastoreapp/storetype/PreferencesDataStore.kt]: Fixed!

File: app/src/main/java/com/example/datastoreapp/viewmodel/PreferencesDataStoreViewModel.kt (hunk 8 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/datastoreapp/viewmodel/PreferencesDataStoreViewModel.kt]: Fixed!

File: app/src/main/java/com/example/datastoreapp/storetype/ProtoDataStoreSerializer.kt (hunk 6 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/datastoreapp/storetype/ProtoDataStoreSerializer.kt:1]: I tried to configure the Indentation but in the code I don't see these issues. Maybe it's a indentation difference between the OS (I have windows)
image

File: app/src/main/java/com/example/datastoreapp/UserInfo.kt (hunk 5 above)
Comment (SofiaZarzabal) [reply app/src/main/java/com/example/datastoreapp/UserInfo.kt:1]: I tried to configure the Indentation but in the code I don't see these issues. Maybe it's a indentation difference between the OS (I have windows)
image

**Review by SartiMau — APPROVED** [review]

**Review by FranciscoGaraventa — APPROVED** [review]
LG

**Pull request merged**